### PR TITLE
Hardware SPI library

### PR DESCRIPTION
**Pull request by mikee47**
Class-based asynchronous SPI driver stack for Sming.
Currently ESP8266-only, but will be extended to support ESP32.
Also contains demo for S1D13781 TFT controller.

### CODE REVIEW

File: samples/ScreenTFT_S1D13781/app/app.cpp [hunk 1]
```diff
@@ -0,0 +1,287 @@
+#include <SmingCore.h>
```
Comment (slaff) [inline samples/ScreenTFT_S1D13781/app/app.cpp:1]: - Please rename this file to app/application.cpp to be consistent with the other samples that we have.
- Remove the anonymous namespace. Adding an anonymous namespace to all samples should happen in a later release.

Comment (mikee47) [reply samples/ScreenTFT_S1D13781/app/app.cpp]: OK, I've moved the sample into the library and stripped out all the RAM stuff so it's much simpler.

**Comment by slaff** [comment]
@mikee47 Ready with this PR?

**Comment by mikee47** [comment]
@slaff yep, thanks.

**Pull request merged**